### PR TITLE
feat: Support conversion of Datastore and MessageTransform between wayflow and agent-spec

### PR DESCRIPTION
**Pull request by youssef62**
**Context**

In [feat: Datastores and MessageTransforms in agent-spec](https://github.com/oracle/agent-spec/pull/38), we introduced **agent-spec** components for Datastores ( `OracleDatabaseDatastore`, `PostGresDatabaseDatastore` and more) and `MessageTransform`s (`MessageSummarizationTransform`, `ConversationSummarizationTransform`). In this PR, we mainly adapt wayflow converters, so we support the conversion of these components between their **wayflow** and **agent-spec** versions.

 **Changes**

 * remove plugin datastores, and convert to and from agent-spec `Datastore`. 
 * remove `PluginMessageTransform` and replace it with `MessageTransform`.
 * add the ability to convert between agent-spec and wayflow `MessageSummarizationTransform`. Similarly for `ConversationSummarizationTransform`. 
 * adapt wayflow summarization transforms so that no caching happens when `datastore` is `None`.

### CODE REVIEW

File: wayflowcore/src/wayflowcore/transforms/transforms.py [hunk 1]
```diff
@@ -8,13 +8,14 @@
...
+from wayflowcore.component import Component
 from wayflowcore.messagelist import Message, MessageType
-from wayflowcore.serialization.serializer import SerializableCallable, SerializableObject
+from wayflowcore.serialization.serializer import SerializableCallable
 
 logger = logging.getLogger(__name__)
 
 
-class MessageTransform(SerializableCallable, SerializableObject):
+class MessageTransform(SerializableCallable, Component):
```
Comment (youssef62) [inline wayflowcore/src/wayflowcore/transforms/transforms.py:18]: @lfaucon Do you think making `MessageTransform` a `Component` is the way to go? 
From one side we still can convert an agent with transforms between agent-spec and wayflow without it (i.e `tests/agentspec/test_agent_with_transforms`) would continue to work. 
On the other side: 
* We wouldn't be able to convert transforms between agent-spec and wayflow (i.e `tests/agentspec/test_transforms.py` fails)
* More importantly, that would be less consistent. It's more consistent to have an agent-spec component translate to a wayflow `Component`. (practically,  for example maybe someone gives a name to their agent-spec that they would like to also find in the wayflow converted object). Do we care about this point?

File: wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py [hunk 2]
```diff
@@ -1072,19 +1076,19 @@ def _datastore_convert_to_agentspec(
         conversion_context: "WayflowToAgentSpecConversionContext",
         runtime_datastore: RuntimeDatastore,
         referenced_objects: Optional[Dict[str, Any]] = None,
-    ) -> AgentSpecPluginDatastore:
+    ) -> AgentSpecDatastore:
         if isinstance(runtime_datastore, RuntimeInMemoryDatastore):
-            return AgentSpecPluginInMemoryDatastore(
-                name="PluginInMemoryDatastore",
+            return AgentSpecInMemoryDatastore(
+                name="InMemoryDatastore",
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: minor: Looks like we don't correctly save and reuse the name coming from the AgentSpec configuration when translating the datastores from and to AgentSpec

File: wayflowcore/src/wayflowcore/templates/_swarmtemplate.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ def _is_system_reminder(message: "Message") -> bool:
 """Max number of characters in the message header when formatting a Tool Result"""
 
 
-class _ToolRequestAndCallsTransform(MessageTransform, SerializableObject):
+class _ToolRequestAndCallsTransform(MessageTransform):
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/templates/_swarmtemplate.py:115]: question: why this change?

File: wayflowcore/src/wayflowcore/transforms/transforms.py (hunk 1 above)
Comment (lfaucon) [reply wayflowcore/src/wayflowcore/transforms/transforms.py:18]: Yes. That sounds right. That will solve my other comment about losing names of Message Transform when translating from and to AgentSpec

File: wayflowcore/tests/agentspec/test_agent_with_transforms.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# Copyright © 2025 Oracle and/or its affiliates.
```
Comment (lfaucon) [inline wayflowcore/tests/agentspec/test_agent_with_transforms.py]: Should be 2026 or "2025, 2026"

File: wayflowcore/tests/agentspec/test_agent_with_transforms.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    _testing_conversation_summarization_transforms,
+    _testing_message_summarization_transforms,
+    assert_conversation_summarization_transforms_are_equal,
+    assert_message_summarization_transforms_are_equal,
+)
+
+filter_inmemds_warnings = pytest.mark.filterwarnings(f"ignore:{_INMEMORY_USER_WARNING}:UserWarning")
+
+
+def _testing_agents_with_summarization_transforms():
```
Comment (lfaucon) [inline wayflowcore/tests/agentspec/test_agent_with_transforms.py:32]: Why defining this as a method to be called, rather than the slightly more declarative & readable pytest's fixture?

Comment (lfaucon) [reply wayflowcore/tests/agentspec/test_agent_with_transforms.py:32]: I think you can also put the fixtures in the parent file `conftest.py` and use them automatically without importing

Comment (lfaucon) [reply wayflowcore/tests/agentspec/test_agent_with_transforms.py:32]: Pytest's outputs are slightly nicer to read when using fixtures than your own method callables

File: wayflowcore/tests/transforms/test_summarization_transforms.py [hunk 6]
```diff
@@ -1029,6 +1018,42 @@ def test_conversation_summarization_trigger_and_cache_incremental(get_setup):
     assert execute_conversation_check_summarizer_ran(conv, summarization_llm, agent_llm)
 
 
+@pytest.mark.filterwarnings(f"ignore:{_SUMMARIZATION_WARNING_MESSAGE}:UserWarning")
```
Comment (lfaucon) [inline wayflowcore/tests/transforms/test_summarization_transforms.py]: 🤔 If the datastore is None then, there should not be the warrning from the inmemory datastore, right?

File: wayflowcore/src/wayflowcore/templates/_swarmtemplate.py (hunk 3 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/templates/_swarmtemplate.py:115]: Because `Component` inherits already from `SerializableObject` and `MessageTransform` inherits from `Component` now. And `_ToolRequestAndCallsTransform` inherits from `MessageTransform` so it inherits already `SerializableObject` indirectly.

File: wayflowcore/src/wayflowcore/transforms/transforms.py (hunk 1 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/transforms/transforms.py:18]: Thank you @lfaucon . Hey @jschweiz, with @lfaucon we wanted also your confirmation that this is the way.

File: wayflowcore/tests/transforms/test_summarization_transforms.py (hunk 6 above)
Comment (youssef62) [reply wayflowcore/tests/transforms/test_summarization_transforms.py]: I agree, corrected in [f11fba3](https://github.com/oracle/wayflow/pull/25/commits/f11fba3d7a8936816d43d5e40902b9c6e3fc7909)

File: wayflowcore/tests/agentspec/test_agent_with_transforms.py (hunk 4 above)
Comment (youssef62) [reply wayflowcore/tests/agentspec/test_agent_with_transforms.py]: Good catch, done in [f11fba3](https://github.com/oracle/wayflow/pull/25/commits/f11fba3d7a8936816d43d5e40902b9c6e3fc7909).

File: wayflowcore/tests/agentspec/test_agent_with_transforms.py (hunk 5 above)
Comment (youssef62) [reply wayflowcore/tests/agentspec/test_agent_with_transforms.py:32]: I made it in a fixture [8cefc9f](https://github.com/oracle/wayflow/pull/25/commits/8cefc9fb5ce4ffb5fce89347e4d3206ee336b743).

File: wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py (hunk 2 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: Currently the wayflow `Datastore` does not have a `name` field. Should we make `Datastore` also a `Component` too?

Note that my question in the other comment was about making `MessageTransform` a Component not `Datastore`.

Comment (youssef62) [reply wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: We agreed to make the datastore into a component for the same reasons as for `MessageTransform`.

This is now corrected in [69ace9b](https://github.com/oracle/wayflow/pull/25/commits/69ace9b749cace891464fcc19d51a78fdec7589c).

File: docs/wayflowcore/source/core/changelog.rst [hunk 7]
```diff
@@ -91,6 +91,11 @@ Improvements
 
   Transforms can now be assigned to an agent through its constructor. (This is not supported in `agentspec`, so attempting to convert agents with transforms to `agentspec` will raise a `NotImplementedError`)
 
+* **Message Summarization Transforms and Datastores can be converted between Agent Spec and Wayflow**
+
+Added support for converting `MessageSummarizationTransform` and `ConversationSummarizationTransform` between Agent Spec and Wayflow. Similarly for Datastores (`OracleDatabaseDatastore`, `PostgreSQLDatabaseDatastore`).
+You can now declare your agents with summarization transforms and summary caching in Agent Spec and run them in WayFlow.
```
Comment (lfaucon) [inline docs/wayflowcore/source/core/changelog.rst]: 🤔 We have a versionning problem here too. This PR depends on Agent Spec 26.2.0, so the PR should be merged after the 26.1.0 release and the update to the changelog should be in the future added version 26.2.0

File: wayflowcore/src/wayflowcore/agent.py [hunk 8]
```diff
@@ -127,7 +127,7 @@ def __init__(
         name: Optional[str] = None,
         description: str = "",
         agent_template: Optional[PromptTemplate] = None,
-        transforms: Optional[List[MessageTransform]] = None,
+        transforms: List[MessageTransform] = [],
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/agent.py]: Please check this article about mutable default values: https://www.geeksforgeeks.org/python/use-mutable-default-value-as-an-argument-in-python/

Then don't use mutable default values unless that's really what you intend to do ^^

File: wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py [hunk 9]
```diff
@@ -1651,8 +1663,9 @@ class SupportsTimeoutKwargs(TypedDict, total=False):
                 connection_config=conversion_context.convert(
                     agentspec_component.connection_config, tool_registry, converted_components
                 ),
+                name=agentspec_component.name,
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py]: Why not:

```suggestion
                **self._get_component_arguments(agentspec_component),
```

like the other components?

File: wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py [hunk 10]
```diff
@@ -1672,6 +1685,47 @@ class SupportsTimeoutKwargs(TypedDict, total=False):
                 return RuntimeReactMergeToolRequestAndCallsTransform()
             elif isinstance(agentspec_component, AgentSpecPluginSwarmToolRequestAndCallsTransform):
                 return RuntimeSwarmToolRequestAndCallsTransform()
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py]: Given that you are changing the parent type `RuntimeMessageTransfom` to be a `wayflowcore.component`, I believe you should also pass the arguments `**self._get_component_arguments(agentspec_component)` in all of the above, no?

File: wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py [hunk 11]
```diff
@@ -354,7 +351,9 @@
 from wayflowcore.models.ociclientconfig import (
     OCIClientConfigWithUserAuthentication as RuntimeOCIClientConfigWithUserAuthentication,
 )
-from wayflowcore.models.openaicompatiblemodel import EMPTY_API_KEY
+from wayflowcore.models.openaicompatiblemodel import (
+    EMPTY_API_KEY,
+)
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: 🤔 Did your dev setup automatically apply this change? I'd expect a small chance that it breaks the CI, it seems to go against the rules set in our pre-commit hook.

File: wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py [hunk 12]
```diff
@@ -1845,8 +1876,22 @@ def _agent_convert_to_agentspec(
...
+        ]
         metadata = _create_agentspec_metadata_from_runtime_component(runtime_agent)
         extended_agent_model_fields = AgentSpecExtendedAgent.model_fields
+        transforms = [
+            cast(
+                AgentSpecMessageTransform,
+                conversion_context.convert(transform, referenced_objects),
+            )
+            for transform in runtime_agent.transforms
+        ]
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: This code block appears twice. Looks like a mistake?

File: wayflowcore/src/wayflowcore/transforms/summarization.py [hunk 13]
```diff
@@ -63,18 +101,10 @@ def __init__(
...
-        else:
-            with warnings.catch_warnings():
-                warnings.filterwarnings("ignore", message=f"{_INMEMORY_USER_WARNING}*")
-                self.datastore = InMemoryDatastore(
-                    self._get_cache_schema(collection_name, entity_def)
-                )
-            warnings.warn(_SUMMARIZATION_WARNING_MESSAGE)
+        self.datastore = datastore
+        # Validate that the user provided datastore corresponds has the required
+        # fields.
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/transforms/summarization.py]: typo:
```suggestion
        # Validate that the user provided datastore has the required fields.
```

File: wayflowcore/src/wayflowcore/transforms/summarization.py [hunk 14]
```diff
@@ -282,23 +314,40 @@ def __init__(
             "Your response will replace the message, so just output the summary directly, no introduction needed."
         ),
         summarized_message_template: str = "Summarized message: {{summary}}",
-        datastore: Optional["Datastore"] = None,
+        datastore: Optional["Datastore"] = _DEFAULT_MESSAGE_DATASTORE,
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/transforms/summarization.py]: We have the same problem here about mutable defaults, and.... that makes me realize one edge case that may be under-tested.

When creating several message transforms they all share the same, unique & mutable, default datastore

Small test to reproduce the issue

```
from wayflowcore.transforms import MessageSummarizationTransform

a = MessageSummarizationTransform(llm="some_llm")
b = MessageSummarizationTransform(llm="some_llm")

assert a.cache.datastore is not b.cache.datastore
```

I'd recommend changing that to have

```
# (1)
class UnspecifiedDatastore:
    pass

# (2)
_DEFAULT_MESSAGE_DATASTORE = UnspecifiedDatastore()

# (3)
if isinstance(datastore, UnspecifiedDatastore):
    datastore = _default_message_datastore()  <--- here you are creating a new object, so each MessageTransform will have it's own default datastore
```

Comment (lfaucon) [reply wayflowcore/src/wayflowcore/transforms/summarization.py]: Then the thing that I mention we have probably under-tested is the edgecase when multiple Agents share the same caching datastore. One thing that would be weird/not working as expected would be when there are conflicting cache retention configurations. The agent configured to more often delete cache entries would be deleting it for all agents.

It's not necessarily a big problem to solve. If developers have this issue with caching, they will easily be able to alogn the cache retention configuration or to use separate caches for separate use-cases.

File: wayflowcore/src/wayflowcore/transforms/summarization.py [hunk 15]
```diff
@@ -425,12 +474,20 @@ def __init__(
         summarization_instructions: str = "Please make a summary of this conversation. Include relevant information and keep it short. "
         "Your response will replace the messages, so just output the summary directly, no introduction needed.",
         summarized_conversation_template: str = "Summarized conversation: {{summary}}",
-        datastore: Optional["Datastore"] = None,
+        datastore: Optional["Datastore"] = _DEFAULT_CONVERSATION_DATASTORE,
```
Comment (lfaucon) [inline wayflowcore/src/wayflowcore/transforms/summarization.py]: Same mutable default issue

File: wayflowcore/tests/transforms/test_summarization_transforms.py [hunk 16]
```diff
@@ -1029,6 +1069,41 @@ def test_conversation_summarization_trigger_and_cache_incremental(get_setup):
...
 
 
+@pytest.mark.parametrize(
+    "transform_type,params",
+    [
+        (MessageSummarizationTransform, {"max_message_size": 500}),
+        (ConversationSummarizationTransform, {"max_num_messages": 10, "min_num_messages": 5}),
+    ],
+)
+def test_summarization_transform_no_caching_when_datastore_none(transform_type, params):
```
Comment (lfaucon) [inline wayflowcore/tests/transforms/test_summarization_transforms.py]:
```suggestion
def test_summarization_transform_does_not_cache_when_datastore_none(transform_type, params):
```

File: wayflowcore/tests/transforms/conftest.py [hunk 17]
```diff
@@ -8,7 +8,17 @@
 from unittest.mock import AsyncMock, patch
 
 import pytest
+from pyagentspec.agent import Agent as AgentSpecAgent
+from pyagentspec.datastores.datastore import InMemoryCollectionDatastore
+from pyagentspec.llms import VllmConfig
```
Comment (lfaucon) [inline wayflowcore/tests/transforms/conftest.py]: minor:
```suggestion
from pyagentspec.datastores.datastore import InMemoryCollectionDatastore as AgentSpecInMemoryCollectio
from pyagentspec.llms import VllmConfig as AgentSpecVllmConfig
```

That helps for readability, especially because other imports follow this principle, so it's surprising if some imports in the same file don't

Comment (youssef62) [reply wayflowcore/tests/transforms/conftest.py]: I agree, done in [6b793d2](https://github.com/oracle/wayflow/pull/25/commits/6b793d20ba802ddb58cfc9d17ba4202908ac2fc6)

File: wayflowcore/src/wayflowcore/transforms/summarization.py (hunk 14 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/transforms/summarization.py]: I agree. Our current implementation is more oriented for having one datastore for each each transform (or at the very least one db table for each transform). Should we make that clear somewhere ?

File: wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py (hunk 9 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py]: Done in [ff13d71](https://github.com/oracle/wayflow/pull/25/commits/ff13d719948bb046f9a5f889551defc09124e15b)

File: wayflowcore/src/wayflowcore/agent.py (hunk 8 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/agent.py]: Thanks for telling me about this. I did not know about it. Corrected here [c9067dc](https://github.com/oracle/wayflow/pull/25/commits/c9067dc54b42a7e96950c9c68a6c6e773040f722).

File: wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py (hunk 10 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/serialization/_builtins_deserialization_plugin.py]: Good catch. Added here [bb53c29](https://github.com/oracle/wayflow/pull/25/commits/bb53c294a23a18111925d1beb0c5e54894fa4cc7).

File: wayflowcore/src/wayflowcore/transforms/summarization.py (hunk 14 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/transforms/summarization.py]: For the mutable defaults, it's fixed in [6764ed1](https://github.com/oracle/wayflow/pull/25/commits/6764ed1e028177e2c5a30a2293d97e4d4a32296d).

File: wayflowcore/src/wayflowcore/transforms/summarization.py (hunk 15 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/transforms/summarization.py]: Fixed in [6764ed1](https://github.com/oracle/wayflow/pull/25/commits/6764ed1e028177e2c5a30a2293d97e4d4a32296d)

File: wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py (hunk 11 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: Yes isort on my precommit hooks does this.

File: wayflowcore/src/wayflowcore/agentspec/components/transforms.py [hunk 18]
```diff
@@ -1,36 +1,27 @@
-# Copyright © 2025 Oracle and/or its affiliates.
+# Copyright © 2026 Oracle and/or its affiliates.
```
Comment (geeooleea) [inline wayflowcore/src/wayflowcore/agentspec/components/transforms.py]: Should this be
```suggestion
# Copyright © 2025, 2026 Oracle and/or its affiliates.
```
instead?

File: wayflowcore/src/wayflowcore/transforms/summarization.py [hunk 19]
```diff
@@ -282,23 +281,47 @@ def __init__(
             "Your response will replace the message, so just output the summary directly, no introduction needed."
         ),
         summarized_message_template: str = "Summarized message: {{summary}}",
-        datastore: Optional["Datastore"] = None,
+        datastore: Union["Datastore", _UnspecifiedDatastore, None] = _DEFAULT_MESSAGE_DATASTORE,
```
Comment (geeooleea) [inline wayflowcore/src/wayflowcore/transforms/summarization.py:287]: This looks like a hidden mutable default. And while not strictly problematic in this case with an empty class, I think it is not so safe to have here either. May I suggest replacing this with an Enum instead? e.g., `UnspecifiedDatastore.DEFAULT_VALUE`

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil

File: wayflowcore/src/wayflowcore/transforms/summarization.py [hunk 20]
```diff
@@ -282,23 +281,47 @@ def __init__(
...
+
+        if isinstance(datastore, _UnspecifiedDatastore):
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", message=f"{_INMEMORY_USER_WARNING}*")
+                datastore = InMemoryDatastore(
+                    {self.cache_collection_name: self.get_entity_definition()}
+                )
+            warnings.warn(_SUMMARIZATION_WARNING_MESSAGE)
+
+        if datastore is not None:
```
Comment (geeooleea) [inline wayflowcore/src/wayflowcore/transforms/summarization.py:320]: Should this change in behaviour be listed in the changelog as a possibly breaking change?

File: wayflowcore/src/wayflowcore/transforms/summarization.py (hunk 19 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/transforms/summarization.py:287]: Done in 6ae2a6db.

File: wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py (hunk 11 above)
Comment (jschweiz) [reply wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: this makes the CI crash, no? For some reason you have to remove this change because the static checks in the CI do not like it

File: wayflowcore/src/wayflowcore/templates/_managerworkerstemplate.py [hunk 21]
```diff
@@ -98,7 +98,7 @@
 """Max number of characters in the message header when formatting a Tool Result"""
 
 
-class _ToolRequestAndCallsTransform(MessageTransform, SerializableObject):
+class _ToolRequestAndCallsTransform(MessageTransform):
```
Comment (jschweiz) [inline wayflowcore/src/wayflowcore/templates/_managerworkerstemplate.py:96]: does it register it automatically?

File: wayflowcore/src/wayflowcore/transforms/summarization.py (hunk 19 above)
Comment (jschweiz) [reply wayflowcore/src/wayflowcore/transforms/summarization.py:287]: what was the problem with just having `Optinal["Datastore"]`?

Comment (jschweiz) [reply wayflowcore/src/wayflowcore/transforms/summarization.py:287]: ah, you wanted to separate `No caching` from `no datastore specified, so defaults to in-memory`

File: wayflowcore/tests/agentspec/test_agentspec_conversion_coverage.py [hunk 22]
```diff
@@ -105,6 +112,10 @@
     "DummyModel",
     "SleepStep",
     "DoNothingStep",
+    # internal transforms not supported in AgentSpec
```
Comment (jschweiz) [inline wayflowcore/tests/agentspec/test_agentspec_conversion_coverage.py:115]: don't we have/need plugins for them at least?

File: wayflowcore/tests/agentspec/test_agent_with_transforms.py [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+
+from ..conftest import MOCK_LLM_CONFIG
+from .test_transforms import (
+    _testing_conversation_summarization_transforms,
+    _testing_message_summarization_transforms,
+    assert_conversation_summarization_transforms_are_equal,
+    assert_message_summarization_transforms_are_equal,
+)
+
+filter_inmemds_warnings = pytest.mark.filterwarnings(f"ignore:{_INMEMORY_USER_WARNING}:UserWarning")
```
Comment (jschweiz) [inline wayflowcore/tests/agentspec/test_agent_with_transforms.py]: nit: use the full name, it's more understandable and not that much less convenient `filter_in_memory_datastore_warnings`

File: wayflowcore/tests/transforms/test_summarization_transforms.py [hunk 24]
```diff
@@ -405,6 +379,170 @@ def test_transform_summarizes_long_messages_only(
...
+    # We have this check because this test requires short and long messages.
+    assert len(long_messages) == 2 and len(short_messages) >= 3
+    # We pass a mock_llm because we will not need the agent's generation.
+    agent_llm = mock_llm()
+    agent = Agent(llm=agent_llm, tools=[], transforms=transforms)
+    check_transform_summarizes_long_messages_only(agent, messages, long_msgs_indices)
+
+
+@retry_test(max_attempts=4)
+@pytest.mark.filterwarnings(f"ignore:{_INMEMORY_USER_WARNING}:UserWarning")
```
Comment (jschweiz) [inline wayflowcore/tests/transforms/test_summarization_transforms.py]: nit: you can share the warning filter mentioned in the previous comment everywhere, and just import it?

**Review by jschweiz — APPROVED** [review]

File: wayflowcore/tests/transforms/test_summarization_transforms.py (hunk 24 above)
Comment (youssef62) [reply wayflowcore/tests/transforms/test_summarization_transforms.py]: Done in [a221d38](https://github.com/oracle/wayflow/pull/25/commits/a221d384413ac1dca2fc1049c04895cea7bc174c).

File: wayflowcore/tests/agentspec/test_agent_with_transforms.py (hunk 23 above)
Comment (youssef62) [reply wayflowcore/tests/agentspec/test_agent_with_transforms.py]: Done in [a221d38](https://github.com/oracle/wayflow/pull/25/commits/a221d384413ac1dca2fc1049c04895cea7bc174c).

File: wayflowcore/tests/agentspec/test_agentspec_conversion_coverage.py (hunk 22 above)
Comment (youssef62) [reply wayflowcore/tests/agentspec/test_agentspec_conversion_coverage.py:115]: * For `_PythonMergeToolRequestAndCallsTransform`, we have no plugin.
* For `_ToolRequestAndCallsTransform`, we have no plugin for the version from `wayflowcore.templates._managerworkerstemplate`. This is a little confusing because there's also a transform with the same name in `wayflowcore/src/wayflowcore/templates/_swarmtemplate.py`. 
* For `SplitPromptOnMarkerMessageTransform`, we have a plugin, I removed it from the list in [b34c9f8](https://github.com/oracle/wayflow/pull/25/commits/b34c9f8a9f818eddaa98ac08cf362e73df8bc24a). Thank you for noticing this.

File: wayflowcore/src/wayflowcore/templates/_managerworkerstemplate.py (hunk 21 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/templates/_managerworkerstemplate.py:96]: What do you mean by "register"?

File: wayflowcore/src/wayflowcore/transforms/summarization.py (hunk 20 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/transforms/summarization.py:320]: Added here [6da5215](https://github.com/oracle/wayflow/pull/25/commits/6da5215f4f91268764b32335faff8b58c00c2dc7)

Comment (youssef62) [reply wayflowcore/src/wayflowcore/transforms/summarization.py:320]: I agree, added this in [6da5215](https://github.com/oracle/wayflow/pull/25/commits/6da5215f4f91268764b32335faff8b58c00c2dc7).

File: docs/wayflowcore/source/core/changelog.rst (hunk 7 above)
Comment (youssef62) [reply docs/wayflowcore/source/core/changelog.rst]: Done here [8e4d54c](https://github.com/oracle/wayflow/pull/25/commits/8e4d54cddfb1c28d1af05e889dc6222ba2d04040)

**Comment by nasseur** [comment]
Merge Gate failed: Build ID #2161

File: wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py (hunk 12 above)
Comment (youssef62) [reply wayflowcore/src/wayflowcore/serialization/_builtins_serialization_plugin.py]: Corrected here [d54ed46](https://github.com/oracle/wayflow/pull/25/commits/d54ed46ea6dff599a2d046d8537887445e527206).

**Comment by nasseur** [comment]
Internal regression succeeded 🍏: Build ID #568
